Source/Details/ASTableLayoutController.m

```diff
@@ -32,169 +30,45 @@ - (instancetype)initWithTableView:(UITableView *)tableView
-   * - That point is beyond the start/end of the table content. (Handled above)
-   *
-   * Solution: Make a search rect, and if we don't
-   * find any rows, keep doubling its height and searching again. In practice,
-   * this will virtually always find a row on the first try (unless you have a 
-   * tall section header and we land near the middle.)
-   */
-  NSIndexPath *result = nil;
-  for (CGRect searchRect = [ASTableLayoutController initialSearchRectDownward:downward targetY:targetY];
-       // continue while result is nil
-       result == nil;
-       // grow search rect after each loop
-       searchRect = [ASTableLayoutController growSearchRect:searchRect downward:downward]) {
-    NSArray *rows = [_tableView indexPathsForRowsInRect:searchRect];
-    if (downward) {
-      result = [ASTableLayoutController fastArrayMin:rows];
-    } else {
-      result = [ASTableLayoutController fastArrayMax:rows];
-    }
-  }
-  return result;
+  *outVisible = [self visibleIndexPaths];
+  *outDisplay = [self indexPathsForScrolling:scrollDirection rangeMode:rangeMode rangeType:ASLayoutRangeTypeDisplay];
+  *outPreload = [self indexPathsForScrolling:scrollDirection rangeMode:rangeMode rangeType:ASLayoutRangeTypePreload];
 }
 
-- (NSSet<NSIndexPath *> *)indexPathsFromIndexPath:(NSIndexPath *)startIndexPath toIndexPath:(NSIndexPath *)endIndexPath
-{
-  ASDisplayNodeAssert([startIndexPath compare:endIndexPath] != NSOrderedDescending, @"Index paths must be in nondescending order. Start: %@, end %@", startIndexPath, endIndexPath);
-
-  NSMutableSet *result = [NSMutableSet set];
-  NSInteger const endSection = endIndexPath.section;
-  NSInteger i = startIndexPath.row;
-  for (NSInteger s = startIndexPath.section; s <= endSection; s++) {
-    // If end section, row <= end.item. Otherwise (row <= sectionRowCount - 1).
-    NSInteger const rowLimit = (s == endSection ? endIndexPath.row : ([_tableView numberOfRowsInSection:s] - 1));
-    for (; i <= rowLimit; i++) {
-      [result addObject:[NSIndexPath indexPathForRow:i inSection:s]];
-    }
-    i = 0;
-  }
-  return result;
-}
-
-- (nullable NSIndexPath *)firstIndexPathInTableView
-{
-  NSInteger sectionCount = _tableView.numberOfSections;
-  for (NSInteger s = 0; s < sectionCount; s++) {
-    if ([_tableView numberOfRowsInSection:s] > 0) {
-      return [NSIndexPath indexPathForRow:0 inSection:s];
-    }
-  }
-  return nil;
-}
-
-- (nullable NSIndexPath *)lastIndexPathInTableView
-{
-  NSInteger lastSectionWithAnyRows = _tableView.numberOfSections;
-  NSInteger rowCount = 0;
-  while (rowCount == 0) {
-    lastSectionWithAnyRows -= 1;
-    if (lastSectionWithAnyRows < 0) {
-      return nil;
-    }
-    rowCount = [_tableView numberOfRowsInSection:lastSectionWithAnyRows];
-  }
-  return [NSIndexPath indexPathForRow:rowCount - 1 inSection:lastSectionWithAnyRows];
-}
-
-// Same as valueForKeyPath:@"@min.self" but faster
-+ (nullable id)fastArrayMin:(NSArray *)array ASDISPLAYNODE_CONST
```

Comment:
@Adlai-Holler Could you add some diff comments to this change?  It's not immediately apparent why so much code is deleted here, and whether you were able to confirm it didn't regress performance to do so.  Since this code works, and is essentially end-of-life'd / slated for replacement soon, I think we should leave it in place unless the replacement has parity-or-better performance.